s3fs/core.py

```diff
@@ -1282,16 +1284,20 @@ async def _info(self, path, bucket=None, key=None, refresh=False, version_id=Non
                     **version_id_kw(version_id),
                     **self.req_kw,
                 )
-                return {
+                info = {
                     "ETag": out.get("ETag", ""),
                     "LastModified": out["LastModified"],
                     "size": out["ContentLength"],
-                    "name": "/".join([bucket, key]),
+                    "Size": out["ContentLength"],
+                    "name": path,
                     "type": "file",
+                    "Key": path,
                     "StorageClass": out.get("StorageClass", "STANDARD"),
                     "VersionId": out.get("VersionId"),
                     "ContentType": out.get("ContentType"),
                 }
+                self.dircache[path] = [info]
```

Comment:
This is unfortunately incorrect. The dircache should be `{path: [list-of-info]}` where `path` is the parent containing the list of files with info. Therefore, although your method successfully caches the detail of the one file of interest, it very probably would have consequences for other operations on the same path or the parent.

Reply:
Do you have a suggestion how to resolve this issue? Assigning the `[info]` to `self._parent(path)` doesn't seem to be a good idea because we did not actually list that directory and it could contain additional files. I actually think there is little harm in doing it in the way I proposed.

Reply:
I am not totally convinced that it's worth the extra effort to prevent multiple HEAD calls. One auth has happened and a session is open, it should always be fast. Are you experiencing a high cost from it?

Reply:
> every range request (e.g., for an individual column) will trigger a head_object request

Ah, this is annoying, and I suppose this is the real problem. When making a range request, it shouldn't need to call `open()` at all, but use `cat`! I don't suppose we can change this upstream.

Reply:
Are you aware of https://developer.nvidia.com/blog/optimizing-access-to-parquet-data-with-fsspec/ , which should let you prospectively download and cache all the pieces you will be needing in one shot, and then arrow effectively loads from memory/local.

Reply:
I'll take a look. Thanks for sharing! The issue with `blockcache` seems to stem from https://github.com/fsspec/filesystem_spec/blob/2fbe8deff1c8d79f3125def9d930d19c213f03f6/fsspec/spec.py#L1605 where it requires the `size` property to create its cache.

Reply:
As a side issue, and sorry that I indulge myself: fastparquet might do a better job that arrow. I find, after all these years, it's still surprisingly often the better performing library, assuming you want pandas rather than actual arrow structures.

Reply:
Appreciate your advice! We are actually trying to move away from pandas towards polars so arrow is preferred. I'll give it a shot though!